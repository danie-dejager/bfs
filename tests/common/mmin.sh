cd "$TEST"

now=$(epoch_time)

"$XTOUCH" -mt "@$((now - 60 * 60))" one_hour_ago
"$XTOUCH" -mt "@$((now - 121))" two_minutes_ago
"$XTOUCH" -mt "@$((now - 61))" one_minute_ago
"$XTOUCH" -mt "@$((now - 30))" thirty_seconds_ago
"$XTOUCH" -mt "@$((now + 60))" in_one_minute
"$XTOUCH" -mt "@$((now + 60 * 60))" in_one_hour

bfs_diff -mindepth 1 \
    \( -mmin -1 -exec printf -- '-mmin -1: %s\n' {} \; -o -true \) \
    \( -mmin  1 -exec printf -- '-mmin  1: %s\n' {} \; -o -true \) \
    \( -mmin +1 -exec printf -- '-mmin +1: %s\n' {} \; -o -true \)
