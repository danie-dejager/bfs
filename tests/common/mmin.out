-mmin  1: ./one_minute_ago
-mmin +1: ./one_hour_ago
-mmin +1: ./two_minutes_ago
-mmin -1: ./in_one_hour
-mmin -1: ./in_one_minute
-mmin -1: ./thirty_seconds_ago
